rfcs/2022-07-28-13691-registered-internal-events.md

```diff
@@ -198,6 +208,18 @@ handle.emit(ByteSize(received.len()));
 - This increases the complexity of the internal event interface, presenting two different interfaces
   for internal events.
 
+- Internal events that can be registered are necessarily distinct from existing events that can be
+  simply emitted. That is, you cannot `register!(BytesSent { … })`, nor can you
+  `emit!(RegisteredBytesSent { … })` (although the latter could be modified to allow
+  `emit!(RegisteredBytesSent { … }, ByteSize(…))`).
+
+- The syntax for naming the handle for storing it in structures is awkward. Writing a macro to
+  handle this is probably overkill. This is particularly awkward if the registration struct uses a
+  lifetime, since this then requires naming that lifetime to access the handle type, which in turn
+  requires a named lifetime bound on the containing structure _even if the handle itself doesn't
+  require one_. This can be avoided by using the handle name itself, but that requires additional
+  knowledge of the internal event details.
```

Comment:
As an aside: we can hopefully solve this in the future by changes I have planned for `metrics` to support both inlined strings (small string optimization) and `Arc<T>`-wrapped strings, allowing for more flexible ways to share labels.